### ISSUE TITLE
Blocktype select dropdowns appear below/under modals. 

### ISSUE THREAD

**Issue by praxiz** [mdx-editor/editor#276]
First of all, love the editor, but I would like to use it in a Dialog (Primereact), but when I hit the Block type dropdown, the dropdown contents appear under the modal. Here is an example, where I've made the modal exceptionally short: 

![image](https://github.com/mdx-editor/editor/assets/4617969/3e4ffe23-81cd-4e52-8edc-1be47cc5f905)

**Comment by praxiz** [comment]
After a bit of fiddling I found the correct value for the correct item. In case anyone is interested, the solution was to change lines 1658 to 1661 in node_modules/@mdxeditor/editor/dist/style.css to:
```
._popupContainer_11eqz_1170 {
  position: relative;
  z-index: 1101;
}

```

**Comment by cyphase** [comment]
_The `.mdxeditor-popup-container` option mentioned below was added about a month **after** this issue was opened/closed, in 7d371825e0e7bcebeefdd47c3de1a67c518fed51._

---

I'm using MDXEditor for the first time, and just ran into this as well with a modal in a different framework. I happened to have skimmed [the Theming docs](https://mdxeditor.dev/editor/docs/theming) not long before, but didn't make the connection at first, until I searched the issue tracker and saw the auto-generated suffix in your previous comment here @praxiz.

You don't need to – and don't want to and shouldn't – modify the CSS file under `node_modules`; among other good reasons, your change will get wiped out on an update. All you need to do is override the z-index of the popup container using the public class name, like so:

```css
.mdxeditor-popup-container {
  z-index: 1101;
  /* Assuming 1101 is the correct value for you. You're probably safe
  enough with any value that gets it to appear above your modal. */
}
```

From [the Theming docs](https://mdxeditor.dev/editor/docs/theming), very lightly edited:

> The structural styling of the component is done through CSS module class names, which have **auto-generated suffixes that can change between versions**. The key DOM elements of the component have **"public" CSS classes that you can safely target with your own CSS**.
> 
>  * mdxeditor - Set at the root element of the editor and the container for the pop-ups/tooltips.
>  * **mdxeditor-popup-container** - assigned to the popup container. Use this if you're **using MDXEditor with a framework like MUI** and you **need to override the z-index of the popups**.

**Comment by petyosi** [comment]
Thank you @cyphase - good to leave such trace here.
